plugin/core/collections.py

```diff
@@ -132,15 +132,15 @@ def update(self, d: Dict[str, Any]) -> None:
             else:
                 self.set(key, value)
 
-    def create_resolved(self, variables: Dict[str, str]) -> "DottedDict":
+    def get_resolved(self, variables: Dict[str, str]) -> Dict[str, Any]:
         """
-        Resolve a DottedDict that may potentially contain template variables like $folder
+        Resolve a DottedDict that may potentially contain template variables like $folder.
 
         :param      variables:  The variables
 
-        :returns:   A copy of this DottedDict, but with the variables replaced
+        :returns:   A copy of underlying dictionary, but with the variables replaced
         """
-        return DottedDict(sublime.expand_variables(self._d, variables))
+        return sublime.expand_variables(self.copy(), variables)
```

Comment:
There is no need to make a copy. `sublime.expand_variables` already does that

```
>>> a={"hello": "$world"}
>>> b=sublime.expand_variables(a, {"world": "foobar"})
>>> a==b
False
>>> a
{'hello': '$world'}
>>> b
{'hello': 'foobar'}
```

Reply:
Yes, was just testing it too. Fixed.